src/main/java/shopping/application/OrderService.java

```diff
@@ -0,0 +1,76 @@
+package shopping.application;
+
+import java.util.List;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import shopping.domain.Cart;
+import shopping.domain.CartItem;
+import shopping.domain.Email;
+import shopping.domain.Order;
+import shopping.domain.User;
+import shopping.dto.request.ExchangeRate;
+import shopping.dto.request.ShoppingPageRequest;
+import shopping.dto.response.OrderResponse;
+import shopping.exception.OrderNotFoundException;
+import shopping.exception.UserNotFoundException;
+import shopping.exception.UserNotMatchException;
+import shopping.repository.CartItemRepository;
+import shopping.repository.OrderRepository;
+import shopping.repository.UserRepository;
+
+@Service
+@Transactional(readOnly = true)
+public class OrderService {
+    
+    private final OrderRepository orderRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+
+    public OrderService(OrderRepository orderRepository, CartItemRepository cartItemRepository,
+            UserRepository userRepository) {
+        this.orderRepository = orderRepository;
+        this.cartItemRepository = cartItemRepository;
+        this.userRepository = userRepository;
+    }
+
+    @Transactional
+    public Long createOrder(String email, ExchangeRate rate) {
+        User user = userRepository.findByEmail(new Email(email))
+                .orElseThrow(() -> new UserNotFoundException(email));
+        List<CartItem> cartItems = cartItemRepository.findAllByUserEmail(new Email(email));
+
+        Order order = saveOrder(rate, user, cartItems);
+        cartItemRepository.deleteAll(cartItems);
+        return order.getId();
+    }
+
+    private Order saveOrder(ExchangeRate rate, User user, List<CartItem> cartItems) {
+        Cart cart = new Cart(cartItems, user);
+        Order order = cart.toOrderWithRatio(rate.toDomain());
+        orderRepository.save(order);
+        return order;
```

Comment:
레포지토리 저장 결과가 저장된 엔티티 기 때문에, 한줄로 줄일 수 있습니다
(그리고 JPA magic으로 인해 두 객체의 참조는 같습니다)

```suggestion
        return orderRepository.save(order);
```